tests/lax_numpy_test.py

```diff
@@ -3905,5 +3906,49 @@ def f(x):
     check_grads(f, (1.,), order=1)
 
 
+  def testWrappedSignaturesMatch(self):
+    """Test that jax.numpy function signatures match numpy."""
+    # Note: testing signatures for an exact match is problematic, because numpy's signatures
+    # evolve from release to release (mainly adding new parameters when necessary), and jax.numpy
+    # adds additional jax-specific parameters to some functions. Because of this, the test checks
+    # that jax.numpy function parameters are a superset of the wrapped numpy function parameters.
+    # That way, when tested against older numpy versions (where functions often have fewer parameters),
+    # the test should pass, and when run against new numpy releases, the tests will fail indicating
+    # where jax functionality must be upgraded to match.
+
+    # TODO(jakevdp): fix the following signatures.
+    skip = {
+      'allclose', 'amax', 'amin', 'angle', 'argmax', 'argmin', 'around', 'broadcast_to',
+      'clip', 'convolve', 'corrcoef', 'correlate', 'cumprod', 'cumproduct', 'cumsum',
+      'diff', 'empty_like', 'einsum', 'einsum_path',
+      'eye', 'full', 'full_like', 'gradient', 'histogram',
+      'isposinf', 'isneginf', 'isscalar', 'max', 'min', 'nancumprod', 'nancumsum',
+      'nanprod', 'nansum', 'ones', 'ones_like', 'pad', 'polyadd', 'polyder', 'polysub',
+      'prod', 'product', 'round', 'stack', 'sum', 'tile', 'zeros_like'
+    }
+    jnp_funcs = {name: getattr(jnp, name) for name in dir(jnp) if name not in skip}
+    func_pairs = {name: (fun, fun.__np_wrapped__) for name, fun in jnp_funcs.items()
+                  if hasattr(fun, '__np_wrapped__')}
+    assert len(func_pairs) > 0
+
+    mismatches = {}
+
+    for name, (jnp_fun, np_fun) in func_pairs.items():
+      # Note: can't use inspect.getfullargspec due to numpy issue
+      # https://github.com/numpy/numpy/issues/12225
```

Comment:
I think `inspect.signature` is a better API, anyways. As the Python docs note:

> Note that signature() and Signature Object provide the recommended API for callable introspection, and support additional behaviours (like positional-only arguments) that are sometimes encountered in extension module APIs. This function is retained primarily for use in code that needs to maintain compatibility with the Python 2 inspect module API.
>
> Changed in version 3.6: This method was previously documented as deprecated in favour of signature() in Python 3.5, but that decision has been reversed in order to restore a clearly supported standard interface for single-source Python 2/3 code migrating away from the legacy getargspec() API.

Reply:
Makes sense, thanks.